### PR TITLE
Feat: Request Permit

### LINKED ISSUE #225: Rerequest payout command
Right now if there was an error on permit generation then there is no way for a bounty hunter to rerequest a payout.

What should be done:
1. Use issue number for nonce on permit generation (to prevent double claims)
2. Add a `/payout` command which regenerates and posts a permit URL

Also check an edge case when bounty bot is set for 2 repositories in a single organization. This way nonces could intersect if the same private key is used for permit generation in 2 repositories. We could use smth like `int(keccak256(${issue.number}${organization.name}${repository.name}))` for the nonce.

### PR DESCRIPTION
**Pull request by Steveantor**
Resolves #225

### CODE REVIEW

File: src/handlers/payout/index.ts [hunk 1]
```diff
@@ -50,12 +50,13 @@ export const handleIssueClosed = async () => {
   }
 
   if (issue.state_reason === StateReason.COMPLETED) {
-    const payoutUrl = await generatePermit2Signature(recipient, priceInEth);
+    const payoutUrl = await generatePermit2Signature(recipient, priceInEth, issue.node_id);
```
Comment (pavlovcik) [inline src/handlers/payout/index.ts:53]: `issue.node_id` looks promising

**Comment by pavlovcik** [comment]
@rndquu @Steveantor can you verify: 
1. If this was implemented correctly? https://github.com/ubiquity/bounty-bot/pull/226#discussion_r1156558148
2. The hypothesis on issue ID staying unchanged even after a repository rename?

Otherwise looks good to me.

**Comment by Steveantor** [comment]
> @rndquu @Steveantor can you verify:
> 
> 1. If this was implemented correctly? [Feat: Request Permit #226 (comment)](https://github.com/ubiquity/bounty-bot/pull/226#discussion_r1156558148)
> 2. The hypothesis on issue ID staying unchanged even after a repository rename?
> 
> Otherwise looks good to me.

I'm using `node_id` which doesn't change regardless of the org renames,
it uses additional measurements like scanning for previous claim comments and `Paid` labels.
@pavlovcik

File: src/helpers/issue.ts [hunk 2]
```diff
@@ -97,6 +97,37 @@ export const getCommentsOfIssue = async (issue_number: number): Promise<Comment[
...
+
+      if (response.data) {
+        response.data.forEach((item) => result.push(item as Comment));
+        page_number++;
+      } else {
+        shouldFetch = false;
+      }
+    }
+  } catch (e: unknown) {
+    shouldFetch = false;
```
[the remark this answers is not in the document]
Comment (Steveantor) [reply src/helpers/issue.ts:126]: the `node_id` is stable and unique now, and I don't think that GitHub would have any merit in changing this behavior, potentially breaking many applications that rely on it.
if the comment fetching fails then the bot would post a warning message (updated),
generally, the comments fetching shouldn't throw even if there are no comments (then it returns [])
@rndquu

File: src/handlers/payout/index.ts [hunk 3]
```diff
@@ -50,12 +50,13 @@ export const handleIssueClosed = async () => {
...
     const tokenSymbol = await getTokenSymbol(paymentToken, rpc);
     const shortenRecipient = shortenEthAddress(recipient, `[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]`.length);
     logger.info(`Posting a payout url to the issue, url: ${payoutUrl}`);
     const comment = `### [ **[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n` + "```" + shortenRecipient + "```";
     await addCommentToIssue(comment, issue.number);
     await deleteLabel(issueDetailed.priceLabel);
+    await addLabelToIssue("Paid");
```
Comment (pavlovcik) [inline src/handlers/payout/index.ts:60]: Why was this added @Steveantor

Comment (Steveantor) [reply src/handlers/payout/index.ts:60]: the paid label indicates that the bounty bot posted the permit link,
it makes the process somewhat faster instead of parsing through all the messages.
